tiledb/api/c_api/profile/profile_api_experimental.h

```diff
@@ -0,0 +1,156 @@
+/**
+ * @file tiledb/api/c_api/profile/profile_api_experimental.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the profile C API for TileDB.
+ */
+
+#ifndef TILEDB_CAPI_PROFILE_EXPERIMENTAL_H
+#define TILEDB_CAPI_PROFILE_EXPERIMENTAL_H
+
+#include "../api_external_common.h"
+#include "../error/error_api_external.h"
+#include "../string/string_api_external.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** C API carrier for a TileDB profile. */
+typedef struct tiledb_profile_handle_t tiledb_profile_t;
+
+/**
+ * Allocates an in-test TileDB profile object.
+ *
+ * @note Directly passing `homedir` is intended primarily for testing purposes,
```

Comment:
I think this note should be a little less prescriptive. I suggest editing just to say that supplying `nullptr` is valid and what happens if the user chooses to do so.